
# Analysis
## Somatic Interactions
```{r,message=FALSE,results='markup'}
try(somaticInteractions(maf = dataForReport[["maf"]], top = 10,genes=dataForReport[["interestedGenes"]], pvalue = c(0.05, 0.1)))
```

<!-- ## Detecting cancer driver genes based on positional clustering -->
```{r,results='hide'}
if (dataForReport[["genome"]] %in% c("hg18","hg19","hg38")) {
  cat("## Detecting cancer driver genes based on positional clustering\n")
  mafOncodrive = oncodrive(maf = dataForReport[["maf"]], AACol = dataForReport[["AACol"]], minMut = 5, pvalMethod = 'zscore')
}
```

```{r}
if (dataForReport[["genome"]] %in% c("hg18","hg19","hg38")) {
  makeDataTable(mafOncodrive[,c("Hugo_Symbol",setdiff(colnames(mafOncodrive),colnames(dataForReport[["maf"]]@gene.summary))),with = FALSE])
  plotOncodrive(res = mafOncodrive, fdrCutOff = 0.05, useFraction = TRUE)
}
```

<!-- ## Summarizing pfam domains -->
```{r}
if (dataForReport[["genome"]] %in% c("hg18","hg19","hg38")) {
  cat("## Summarizing pfam domains\n")
  mafObj=dataForReport[["maf"]]
  #remove not Standard AaChange. Something like "wholegene" will report error. Add ; behind it to aviod it
  mafObj@data[[dataForReport[["AACol"]]]]=gsub("^[A-Za-z]+$","\\1;",dataForReport[["maf"]]@data[[dataForReport[["AACol"]]]])
  mafPfamDomains= pfamDomains(maf = mafObj, AACol = dataForReport[["AACol"]], top = 10)
}
```


<!-- ## Drug-Gene Interactions -->
```{r,results='markup'}
if (dataForReport[["genome"]] %in% c("hg18","hg19","hg38")) {
  cat("## Drug-Gene Interactions\n")
  mafDrugInteractions = drugInteractions(maf = dataForReport[["maf"]], fontSize = 0.75)
}
```



<!-- ## Oncogenic Signaling Pathways -->
```{r,results='markup'}
if (dataForReport[["genome"]] %in% c("hg18","hg19","hg38")) {
  cat("## Oncogenic Signaling Pathways\n")
  mafOncogenicPathways = OncogenicPathways(maf = dataForReport[["maf"]])
plot(mafOncogenicPathways)
PlotOncogenicPathways(maf = dataForReport[["maf"]], pathways = mafOncogenicPathways$data$Pathway[1])
}
```


<!-- ## Tumor heterogeneity and MATH scores -->
<!-- Top 5 most mutated samples were used -->
```{r}
if (dataForReport[["performVaf"]]) {
  cat("## Tumor heterogeneity and MATH scores")
  mafInferHeterogeneity = inferHeterogeneity(maf = dataForReport[["maf"]], vafCol = dataForReport[["vafCol"]])
  
  #makeDataTable(mafInferHeterogeneity$clusterMeans)
  out=list(makeDataTable(mafInferHeterogeneity$clusterMeans))
  print(tagList(out))
  
  plotClusters(clusters = mafInferHeterogeneity)
}

```

<!-- ```{r} -->
<!-- mafInferHeterogeneity = inferHeterogeneity(maf = dataForReport[["maf"]], vafCol = 'i_TumorVAF_WU') -->
<!-- gvisTable(mafInferHeterogeneity$clusterMeans,options=tableVisOptions) -->
<!-- plotClusters(clusters = mafInferHeterogeneity) -->

<!-- ``` -->

## Mutational Signatures

```{r MutationalSignatures,results='markup',eval=FALSE}
if (dataForReport[["genome"]]=="mm10") {
  genomePackage=paste0("BSgenome.Mmusculus.UCSC.",dataForReport[["genome"]])
} else {
  genomePackage=paste0("BSgenome.Hsapiens.UCSC.",dataForReport[["genome"]])
}

library(genomePackage, quietly = TRUE,character.only=TRUE)
#mafTnm = trinucleotideMatrix(maf = dataForReport[["maf"]], prefix = 'chr', add = TRUE, ref_genome =dataForReport[["genomePackage"]])

#check "chr" issue. Please note we changed seqlevelsStyle of maf file. May influence other analysis in future. So need to use another variable
genomePkgStype=suppressWarnings(GenomeInfoDb::seqlevelsStyle(BSgenome::getBSgenome(genome =genomePackage)))
mafObj=dataForReport[["maf"]]
mafObj@data$Chromosome=as.character(mafObj@data$Chromosome)
mafObj@maf.silent$Chromosome=as.character(mafObj@maf.silent$Chromosome)
GenomeInfoDb::seqlevelsStyle(mafObj@data$Chromosome)=genomePkgStype
GenomeInfoDb::seqlevelsStyle(mafObj@maf.silent$Chromosome)=genomePkgStype

mafTnm = trinucleotideMatrix(maf = mafObj,ref_genome =genomePackage)
try(plotApobecDiff(tnm = mafTnm, maf = mafObj))

library('NMF')
mafTnmSign=try(extractSignatures(mat = mafTnm, plotBestFitRes = FALSE))
try(plotSignatures(mafTnmSign, title_size = 1 ,yaxisLim=NA))

```


<!-- ## dndscv -->
```{r dndscv}
if (!is.null(dataForReport[["dndscv.refdb"]])) {
  cat("## dndscv\n")
  
#  library(dndscv)
  mafDatSnp=rbind(dataForReport[["maf"]]@data,dataForReport[["maf"]]@maf.silent)
  #mafDatSnp=mafDatSnp[which(mafDatSnp$Variant_Type=="SNP"),]
  dataForModel=mafDatSnp[,.(Tumor_Sample_Barcode ,Chromosome,Start_Position,Reference_Allele,Tumor_Seq_Allele2)]
  dataForModel$Chromosome=as.character(dataForModel$Chromosome)

  #check chr style in data
  if (dataForReport[["dndscv.refdb"]] == "hg19") {
     data("refcds_hg19", package = "dndscv")
  } else {
     load(dataForReport[["dndscv.refdb"]])
  }
  genomePkgStype=GenomeInfoDb::seqlevelsStyle(sapply(RefCDS,function(x) x$chr))
  GenomeInfoDb::seqlevelsStyle(dataForModel$Chromosome)=genomePkgStype
  #colnames(dataForModel)<-c("sampleID","chr","pos","ref","alt")

  dndsout = dndscv::dndscv(dataForModel,cv=NULL,refdb=dataForReport[["dndscv.refdb"]],use_indel_sites=FALSE)
  #print(names(dndsout))

  makeDataTable(dndsout$sel_cv[1:500,])
#  kable(dndsout$globaldnds,row.names=FALSE)
}

if (!is.null(dataForReport[["dndscv.refdb"]])) {  #have to be in a seprate section to make sure makeDataTable(dndsout$sel_cv[1:500,]) can display
  kable(dndsout$globaldnds,row.names=FALSE)
}
```





